pallets/order-book/src/benchmarking.rs

```diff
@@ -486,10 +486,44 @@ benchmarks! {
     }
 
     exchange {
+        let caller = alice::<T>();
+
+        let order_book_id = OrderBookId::<AssetIdOf<T>> {
+            base: VAL.into(),
+            quote: XOR.into(),
+        };
+
+        create_and_fill_order_book::<T>(order_book_id);
+
+        Assets::<T>::update_balance(
+            RawOrigin::Root.into(),
+            caller.clone(),
+            order_book_id.base,
+            balance!(1000000).try_into().unwrap()
+        ).unwrap();
+
+        let caller_base_balance = <T as Config>::AssetInfoProvider::free_balance(&order_book_id.base, &caller).unwrap();
+        let caller_quote_balance = <T as Config>::AssetInfoProvider::free_balance(&order_book_id.quote, &caller).unwrap();
     }: {
-        // todo (m.tagirov)
+        OrderBookPallet::<T>::exchange(
+            &caller,
+            &caller,
+            &DEX.into(),
+            &VAL.into(),
+            &XOR.into(),
+            SwapAmount::with_desired_output(balance!(3500), balance!(360)),
+        )
+        .unwrap();
     }
     verify {
+        assert_eq!(
+            <T as Config>::AssetInfoProvider::free_balance(&order_book_id.base, &caller).unwrap(),
+            caller_base_balance - balance!(355.13473)
+        );
+        assert_eq!(
+            <T as Config>::AssetInfoProvider::free_balance(&order_book_id.quote, &caller).unwrap(),
+            caller_quote_balance + balance!(3499.999935)
```

Comment:
Maybe use `assert_approx_eq` if it's due to FP math

Reply:
I prefer to work with exact numbers

Reply:
It just seems like some magic number